alchemiscale/compute/api.py

```diff
@@ -92,6 +97,53 @@ async def list_scopes(
     return [str(scope) for scope in scopes]
 
 
+@router.post("/computeservice/{compute_service_id}/register")
+async def register_computeservice(
+    compute_service_id,
+    n4js: Neo4jStore = Depends(get_n4js_depends),
+):
+    now = datetime.utcnow()
+    csreg = ComputeServiceRegistration(
+        identifier=compute_service_id, registered=now, heartbeat=now
+    )
+
+    compute_service_id_ = n4js.register_computeservice(csreg)
+
+    return compute_service_id_
+
+
+@router.post("/computeservice/{compute_service_id}/deregister")
+async def deregister_computeservice(
+    compute_service_id,
+    n4js: Neo4jStore = Depends(get_n4js_depends),
+):
+    compute_service_id_ = n4js.deregister_computeservice(
+        ComputeServiceID(compute_service_id)
+    )
+
+    return compute_service_id_
+
+
+@router.post("/computeservice/{compute_service_id}/heartbeat")
+async def heartbeat_computeservice(
+    compute_service_id,
+    n4js: Neo4jStore = Depends(get_n4js_depends),
+    settings: ComputeAPISettings = Depends(get_base_api_settings),
+):
+    now = datetime.utcnow()
+
+    # expire any stale registrations, along with their claims
+    expire_delta = timedelta(
+        seconds=settings.ALCHEMISCALE_COMPUTE_API_REGISTRATION_EXPIRE_SECONDS
+    )
+    expire_time = now - expire_delta
+    n4js.expire_registrations(expire_time)
```

Comment:
Query: is there any need for this to return or expose the expired `compute_service_id`s? I'll leave it up to you, im sure this is fine for now.

Reply:
Not at the moment no. The approach we take here (stacking in expiration with the heartbeat API call) is actually a bit of an inefficient hack; the better approach to this will be to have a separate `PeriodicsService` that performs operations like this on some schedule, since they only have to happen once in a while. This will avoid thrashing the database.

I'll create an issue articulating the vision for this service so we can implement for a future release.

Reply:
Issue here: https://github.com/openforcefield/alchemiscale/issues/109